src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,838 @@
+          v-model="selectedCalendarRange"
+          :is-range="range"
+          :max-date="maxDate"
+          :min-date="minDate"
+          :mode="mode"
+          :model-config="modelConfig"
+          :minute-increment="minuteIncrement"
+          :select-attribute="calendarSelectAttributes"
+          :drag-attribute="calendarDragAttributes"
+          is-expanded
+        />
+        <div
+          v-else-if="hasTimePeriods"
+          class="d-flex flex-column"
+        >
+          <div
+            v-for="(item, index) in timePeriods"
+            :key="`section-${String(item.section || index)}`"
+            class="timeframe-section d-flex flex-column"
+          >
+            <div class="timeframe-section-title type-sm mt-2 mb-1">
+              {{ item.section }}
+            </div>
+            <div class="timeframe-buttons d-flex">
+              <KButton
+                v-for="(timeFrame, itemIdx) in item.values"
+                :key="`time-${itemIdx}`"
+                :is-rounded="false"
+                :class="{ 'selected-option': timeFrame.key === selectedTimeframe.key }"
+                :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+                class="timeframe-btn"
+                appearance="outline"
+                size="medium"
+                @click="changeRelativeTimeframe(timeFrame)"
+              >
+                {{ ucWord(timeFrame.timeframeText) }}
+              </KButton>
+            </div>
+          </div>
+        </div>
+      </template>
+      <template
+        v-if="!hidePopover"
+        #footer
+      >
+        <div class="d-flex justify-content-end">
+          <KButton
+            :is-rounded="false"
+            data-testid="k-datetime-picker-clear"
+            class="action-btn"
+            size="medium"
+            appearance="btn-link"
+            @click="clearSelection()"
+          >
+            Clear
+          </KButton>
+          <KButton
+            :disabled="submitDisabled"
+            :is-rounded="false"
+            data-testid="k-datetime-picker-submit"
+            class="action-btn"
+            size="medium"
+            appearance="btn-link"
+            @click="submitTimeFrame()"
+          >
+            Apply
+          </KButton>
+        </div>
+      </template>
+    </KPop>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, nextTick, onMounted, PropType, reactive, ref, toRefs, watch } from 'vue'
+import { format } from 'date-fns'
+import 'v-calendar/dist/style.css'
+import { DatePicker } from 'v-calendar'
+import KButton from '@/components/KButton/KButton.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import KSegmentedControl from '@/components/KSegmentedControl/KSegmentedControl.vue'
+
+export interface TimeRange {
+  start: Date | number,
+  end: Date | number,
+  timePeriodsKey?: string | undefined
```

Comment:
You don't need the ` | undefined` here on the type. Since the property is optional, it's implied.

```suggestion
  timePeriodsKey?: string
```